### DISCUSSION TITLE
Dynamic workspaces

### DISCUSSION THREAD

**Discussion by assaf758** [nikitabobko/AeroSpace#782] [feature-ideas]
Suggest to consider [dynamic workspaces](https://github.com/BurntSushi/wingo#workspaces) - where the user can give a workspace a name (and not just one letter/digit).
Workspace selection would be done using fuzzy-search.
Can be helpful (for example )when there is a contention on letter 'C' between 'Code' and 'Chorme'.

Another use case - workspace per project/feature.

Lastly - will occupy less keyboard shortcuts

**Comment by bradediger** [comment]
I'd been missing the same thing, as I used to heavily use topic spaces as part of my wmii workflow on Linux. Since Aerospace workspace names can already be arbitrary strings, this turned out to be closer within reach than I thought. Here's something I wired up to emulate topic spaces using [choose](https://github.com/chipsenkbeil/choose) (`brew install choose-gui`):

```
alt-t = "exec-and-forget aerospace list-workspaces --all | choose -u -m -p 'Switch to:' | xargs aerospace workspace"
alt-shift-t = "exec-and-forget aerospace list-workspaces --all | choose -u -m -p 'Move to:' | xargs aerospace move-node-to-workspace"
```

I've only been living with this casually on my personal laptop for a few days; I've yet to look at what will need to change for my multimonitor work setup. But it's been fairly useful for such a low-effort endeavor!

**Reply by assaf758** [reply]
Nice @bradediger, thanks for sharing!
Actually, I wasn't aware that aerospace supports naming the workspaces - can you please explain how do you do that?

**Reply by bradediger** [reply]
I'm actually not sure whether the behavior is intentional or not -- but it is quite useful! I had noticed a while back that arbitrary workspace names seemed to work just fine with unmodified AeroSpace (for example, `aerospace workspace web`). On Linux with wmii and/or xmonad, I had used `dmenu` as a typeahead-search finder just as `choose` works above. So I was fairly sure I could come up with the same sort of pipeline (show all workspaces | filter with typeahead search | do something with the selected value) but was blocked on not really having a great `dmenu` replacement. I even briefly played around with writing something as an excuse to learn some SwiftUI, but I don't have a ton of free time these days.

When I found `choose` the other day, it struck me as exactly the `dmenu` replacement I'd been looking for. So after a bit of tinkering I arrived at what I posted above. Glad to hear this approach was useful or at least interesting! :-)
